include/VecCore/Backend/Implementation.h

```diff
@@ -284,6 +285,21 @@ VECCORE_FORCE_INLINE VECCORE_ATT_HOST_DEVICE constexpr Bool_s EarlyReturnAllowed
 #endif
 }
 
+// An extension of EarlyReturnAllowed(), and new name is a reminder that it can be used for other purposes as well
+template <typename T, size_t simdSize = 1>
+VECCORE_FORCE_INLINE
+VECCORE_ATT_HOST_DEVICE
+constexpr Bool_s SIMDsizeUpTo()
```

Comment:
@pcanal I personally think that using `EarlyReturnAllowed() && VectorSize<T>() >= n` directly in code is the best thing. It's much clearer than hiding the check within a function. Otherwise, it might make more sense to create a different function named `CheckEarlyReturn()` or similar that augments the conditions checked to a bit more than the hardware features, like a minimum length that is larger than one. This function looks a bit backwards also, since it always returns false on CUDA. For instance, `SIMDsizeUpTo<double, 1>()` should return true everywhere. What we want is to check that the length is wide enough to get gains in vector code.